File: fc-chat/demo/part3/ch4/message-system-add-push/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/CreateRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import net.prostars.messagesystem.constant.MessageType;

public class CreateRequest extends BaseRequest {

  private final String title;
  private final List<String> participantUsernames;

  @JsonCreator
  public CreateRequest(
      @JsonProperty("title") String title,
      @JsonProperty("participantUsernames") List<String> participantUsernames) {
    super(MessageType.CREATE_REQUEST);
    this.title = title;
    this.participantUsernames = participantUsernames;
  }

  public String getTitle() {
    return title;
  }

  public List<String> getParticipantUsernames() {
    return participantUsernames;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/CreateRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType

class CreateRequest @JsonCreator constructor(
    @param:JsonProperty("title") val title: String,
    @param:JsonProperty("participantUsernames") val participantUsernames: List<String>
) : BaseRequest(MessageType.CREATE_REQUEST)
